localstack-core/localstack/services/kms/provider.py

```diff
@@ -1605,6 +1676,20 @@ def _validate_plaintext_key_type_based(
                 f"Algorithm {encryption_algorithm} and key spec {key.metadata['KeySpec']} cannot encrypt data larger than {max_size_bytes} bytes."
             )
 
+    def _validate_key_supports_include_key_material(self, key: KmsKey):
+        if key.metadata["Origin"] != OriginType.AWS_KMS:
+            raise UnsupportedOperationException(
+                f"{key.metadata['Arn']} origin is {key.metadata.get('Origin')} which is not valid for this operation."
+            )
```

Comment:
What happens if the KMS key is imported here, i.e. `Origin` is `EXTERNAL`? 
It should still allow to list key rotations: but rather we can throw a 5xx instead of 4xx and say that its not yet implemented in localstack.